from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_SignUpPage(object):
    def setupUi(self, SignUpPage):
        SignUpPage.setObjectName("SignUpPage")
        SignUpPage.resize(700, 500)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Fixed, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(SignUpPage.sizePolicy().hasHeightForWidth())
        SignUpPage.setSizePolicy(sizePolicy)
        SignUpPage.setMinimumSize(QtCore.QSize(700, 500))
        SignUpPage.setMaximumSize(QtCore.QSize(700, 500))
        self.centralwidget = QtWidgets.QWidget(SignUpPage)
        self.centralwidget.setStyleSheet("QLineEdit {\n"
"    color: white;\n"
"    border-radius: 12px;\n"
"    padding: 0 8px;\n"
"    background-color:rgba(128,128,128,0);\n"
"    selection-background-color: blue;\n"
"}\n"
"\n"
"QPushButton{\n"
"    color: #000000;\n"
"    background-color: rgba(200,255,200, 0.3);\n"
"    border: 3px solid transparent;\n"
"    border-left: 5px transparent;\n"
"    border-right: 5px transparent;\n"
"    border-radius: 14px;\n"
"    min-width: 80px;\n"
"}\n"
"\n"
"QPushButton:pressed {\n"
"    border: 5px solid #0022ff;\n"
"    border-left: 5px transparent;\n"
"    border-right: 5px transparent;\n"
"    color: rgba(100,100,100,1);\n"
"    \n"
"}\n"
"\n"
"QPushButton:hover{\n"
"    border: 3px solid #00ff22;\n"
"    border-left:3px transparent;\n"
"    border-right:3px transparent;\n"
"}\n"
"\n"
"QPushButton:default {\n"
"    border-color: navy; /* make the default button prominent */\n"
"}\n"
" ")
        self.centralwidget.setObjectName("centralwidget")
        self.label_name = QtWidgets.QLabel(self.centralwidget)
        self.label_name.setGeometry(QtCore.QRect(40, 130, 81, 51))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(11)
        font.setBold(True)
        font.setWeight(75)
        self.label_name.setFont(font)
        self.label_name.setObjectName("label_name")
        self.label_email = QtWidgets.QLabel(self.centralwidget)
        self.label_email.setGeometry(QtCore.QRect(40, 210, 321, 51))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_email.setFont(font)
        self.label_email.setObjectName("label_email")
        self.label_password = QtWidgets.QLabel(self.centralwidget)
        self.label_password.setGeometry(QtCore.QRect(40, 290, 161, 51))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_password.setFont(font)
        self.label_password.setObjectName("label_password")
        self.lineEdit_name = QtWidgets.QLineEdit(self.centralwidget)
        self.lineEdit_name.setGeometry(QtCore.QRect(30, 160, 321, 41))
        font = QtGui.QFont()
        font.setFamily("Consolas")
        font.setPointSize(16)
        self.lineEdit_name.setFont(font)
        self.lineEdit_name.setStyleSheet("")
        self.lineEdit_name.setObjectName("lineEdit_name")
        self.lineEdit_email = QtWidgets.QLineEdit(self.centralwidget)
        self.lineEdit_email.setGeometry(QtCore.QRect(30, 240, 321, 41))
        font = QtGui.QFont()
        font.setFamily("Consolas")
        font.setPointSize(16)
        font.setBold(False)
        font.setWeight(50)
        self.lineEdit_email.setFont(font)
        self.lineEdit_email.setStyleSheet("")
        self.lineEdit_email.setPlaceholderText("")
        self.lineEdit_email.setObjectName("lineEdit_email")
        self.lineEdit_password = QtWidgets.QLineEdit(self.centralwidget)
        self.lineEdit_password.setGeometry(QtCore.QRect(30, 320, 321, 41))
        font = QtGui.QFont()
        font.setFamily("Consolas")
        font.setPointSize(16)
        self.lineEdit_password.setFont(font)
        self.lineEdit_password.setStatusTip("")
        self.lineEdit_password.setWhatsThis("")
        self.lineEdit_password.setStyleSheet("")
        self.lineEdit_password.setInputMask("")
        self.lineEdit_password.setText("")
        self.lineEdit_password.setFrame(True)
        self.lineEdit_password.setEchoMode(QtWidgets.QLineEdit.PasswordEchoOnEdit)
        self.lineEdit_password.setCursorMoveStyle(QtCore.Qt.VisualMoveStyle)
        self.lineEdit_password.setObjectName("lineEdit_password")
        self.pushButton_create = QtWidgets.QPushButton(self.centralwidget)
        self.pushButton_create.setGeometry(QtCore.QRect(250, 430, 191, 41))
        font = QtGui.QFont()
        font.setFamily("Segoe UI Semibold")
        font.setPointSize(14)
        font.setBold(True)
        font.setWeight(75)
        self.pushButton_create.setFont(font)
        self.pushButton_create.setObjectName("pushButton_create")
        self.pushButton_back = QtWidgets.QPushButton(self.centralwidget)
        self.pushButton_back.setGeometry(QtCore.QRect(20, 460, 81, 31))
        font = QtGui.QFont()
        font.setFamily("Segoe UI Semibold")
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.pushButton_back.setFont(font)
        self.pushButton_back.setObjectName("pushButton_back")
        self.label_caa = QtWidgets.QLabel(self.centralwidget)
        self.label_caa.setGeometry(QtCore.QRect(20, 20, 271, 41))
        font = QtGui.QFont()
        font.setFamily("Arial")
        font.setPointSize(18)
        font.setBold(True)
        font.setWeight(75)
        self.label_caa.setFont(font)
        self.label_caa.setAlignment(QtCore.Qt.AlignCenter)
        self.label_caa.setObjectName("label_caa")
        self.label_cpassword = QtWidgets.QLabel(self.centralwidget)
        self.label_cpassword.setGeometry(QtCore.QRect(390, 290, 201, 51))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_cpassword.setFont(font)
        self.label_cpassword.setObjectName("label_cpassword")
        self.lineEdit_confpassword = QtWidgets.QLineEdit(self.centralwidget)
        self.lineEdit_confpassword.setGeometry(QtCore.QRect(370, 320, 271, 41))
        font = QtGui.QFont()
        font.setFamily("Consolas")
        font.setPointSize(16)
        self.lineEdit_confpassword.setFont(font)
        self.lineEdit_confpassword.setObjectName("lineEdit_confpassword")
        self.label_dob = QtWidgets.QLabel(self.centralwidget)
        self.label_dob.setGeometry(QtCore.QRect(390, 120, 101, 51))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_dob.setFont(font)
        self.label_dob.setObjectName("label_dob")
        self.label_phone = QtWidgets.QLabel(self.centralwidget)
        self.label_phone.setGeometry(QtCore.QRect(390, 210, 111, 51))
        font = QtGui.QFont()
        font.setFamily("Calibri")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_phone.setFont(font)
        self.label_phone.setObjectName("label_phone")
        self.lineEdit_phone = QtWidgets.QLineEdit(self.centralwidget)
        self.lineEdit_phone.setGeometry(QtCore.QRect(370, 240, 271, 41))
        font = QtGui.QFont()
        font.setPointSize(16)
        font.setBold(False)
        font.setWeight(50)
        self.lineEdit_phone.setFont(font)
        self.lineEdit_phone.setAcceptDrops(False)
        self.lineEdit_phone.setStyleSheet("")
        self.lineEdit_phone.setInputMethodHints(QtCore.Qt.ImhDigitsOnly|QtCore.Qt.ImhFormattedNumbersOnly|QtCore.Qt.ImhPreferNumbers)
        self.lineEdit_phone.setPlaceholderText("")
        self.lineEdit_phone.setCursorMoveStyle(QtCore.Qt.LogicalMoveStyle)
        self.lineEdit_phone.setObjectName("lineEdit_phone")
        self.label_H_archive = QtWidgets.QLabel(self.centralwidget)
        self.label_H_archive.setGeometry(QtCore.QRect(520, 20, 221, 81))
        font = QtGui.QFont()
        font.setFamily("Berlin Sans FB")
        self.label_H_archive.setFont(font)
        self.label_H_archive.setLayoutDirection(QtCore.Qt.LeftToRight)
        self.label_H_archive.setStyleSheet("")
        self.label_H_archive.setAlignment(QtCore.Qt.AlignCenter)
        self.label_H_archive.setObjectName("label_H_archive")
        self.bg_img = QtWidgets.QLabel(self.centralwidget)
        self.bg_img.setGeometry(QtCore.QRect(0, 0, 701, 511))
        self.bg_img.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.bg_img.setLineWidth(1)
        self.bg_img.setMidLineWidth(1)
        self.bg_img.setText("")
        self.bg_img.setTextFormat(QtCore.Qt.RichText)
        #self.bg_img.setPixmap(QtGui.QPixmap("../../../../../WALLZ/earth.jpg"))
        self.bg_img.setPixmap(QtGui.QPixmap("res/earth_b.jpg"))
        self.bg_img.setScaledContents(True)
        self.bg_img.setWordWrap(False)
        self.bg_img.setObjectName("bg_img")
        self.dateEdit = QtWidgets.QDateEdit(self.centralwidget)
        self.dateEdit.setGeometry(QtCore.QRect(390, 170, 181, 41))
        font = QtGui.QFont()
        font.setFamily("Consolas")
        font.setPointSize(14)
        self.dateEdit.setFont(font)
        self.dateEdit.setStyleSheet("color:white;\n"
"background-color: rgba(128,128,128,0.3);\n"
"border-radius: 12px;")
        self.dateEdit.setAlignment(QtCore.Qt.AlignCenter)
        self.dateEdit.setCalendarPopup(True)
        self.dateEdit.setObjectName("dateEdit")
        self.line_n = QtWidgets.QFrame(self.centralwidget)
        self.line_n.setGeometry(QtCore.QRect(40, 190, 321, 20))
        self.line_n.setMidLineWidth(1)
        self.line_n.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_n.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_n.setObjectName("line_n")
        self.line_e = QtWidgets.QFrame(self.centralwidget)
        self.line_e.setGeometry(QtCore.QRect(40, 270, 321, 20))
        self.line_e.setMidLineWidth(1)
        self.line_e.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_e.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_e.setObjectName("line_e")
        self.line_p = QtWidgets.QFrame(self.centralwidget)
        self.line_p.setGeometry(QtCore.QRect(40, 350, 321, 21))
        self.line_p.setStyleSheet("color:black;")
        self.line_p.setFrameShadow(QtWidgets.QFrame.Plain)
        self.line_p.setLineWidth(2)
        self.line_p.setMidLineWidth(1)
        self.line_p.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_p.setObjectName("line_p")
        self.line_pn = QtWidgets.QFrame(self.centralwidget)
        self.line_pn.setGeometry(QtCore.QRect(387, 270, 261, 20))
        self.line_pn.setLineWidth(1)
        self.line_pn.setMidLineWidth(1)
        self.line_pn.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_pn.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_pn.setObjectName("line_pn")
        self.line_cp = QtWidgets.QFrame(self.centralwidget)
        self.line_cp.setGeometry(QtCore.QRect(380, 350, 271, 20))
        self.line_cp.setFrameShadow(QtWidgets.QFrame.Plain)
        self.line_cp.setLineWidth(2)
        self.line_cp.setMidLineWidth(1)
        self.line_cp.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_cp.setObjectName("line_cp")
        self.line_main = QtWidgets.QFrame(self.centralwidget)
        self.line_main.setGeometry(QtCore.QRect(17, 70, 671, 31))
        self.line_main.setFrameShape(QtWidgets.QFrame.HLine)
        self.line_main.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_main.setObjectName("line_main")
        self.label_H_password = QtWidgets.QLabel(self.centralwidget)
        self.label_H_password.setGeometry(QtCore.QRect(530, 10, 141, 41))
        self.label_H_password.setAlignment(QtCore.Qt.AlignCenter)
        self.label_H_password.setObjectName("label_H_password")
        self.lineEdit_errmsg = QtWidgets.QLineEdit(self.centralwidget)
        self.lineEdit_errmsg.setGeometry(QtCore.QRect(130, 380, 431, 41))
        font = QtGui.QFont()
        font.setPointSize(12)
        self.lineEdit_errmsg.setFont(font)
        self.lineEdit_errmsg.setStyleSheet("")
        self.lineEdit_errmsg.setAlignment(QtCore.Qt.AlignCenter)
        self.lineEdit_errmsg.setReadOnly(True)
        self.lineEdit_errmsg.setObjectName("lineEdit_errmsg")
        self.bg_img.raise_()
        self.label_name.raise_()
        self.label_email.raise_()
        self.label_password.raise_()
        self.lineEdit_name.raise_()
        self.lineEdit_email.raise_()
        self.lineEdit_password.raise_()
        self.pushButton_create.raise_()
        self.pushButton_back.raise_()
        self.label_caa.raise_()
        self.label_cpassword.raise_()
        self.lineEdit_confpassword.raise_()
        self.label_dob.raise_()
        self.label_phone.raise_()
        self.lineEdit_phone.raise_()
        self.label_H_archive.raise_()
        self.dateEdit.raise_()
        self.line_n.raise_()
        self.line_e.raise_()
        self.line_p.raise_()
        self.line_pn.raise_()
        self.line_cp.raise_()
        self.line_main.raise_()
        self.label_H_password.raise_()
        self.lineEdit_errmsg.raise_()
        SignUpPage.setCentralWidget(self.centralwidget)

        self.retranslateUi(SignUpPage)
        QtCore.QMetaObject.connectSlotsByName(SignUpPage)

    def retranslateUi(self, SignUpPage):
        _translate = QtCore.QCoreApplication.translate
        SignUpPage.setWindowTitle(_translate("SignUpPage", "MainWindow"))
        self.label_name.setText(_translate("SignUpPage", "<html><head/><body><p><span style=\" font-size:12pt; color:#ffffff;\">Name </span></p></body></html>"))
        self.label_email.setText(_translate("SignUpPage", "<html><head/><body><p><span style=\" font-size:12pt; color:#ffffff;\">Email ID</span></p></body></html>"))
        self.label_password.setText(_translate("SignUpPage", "<html><head/><body><p><span style=\" font-size:12pt; color:#ffffff;\">Password </span></p></body></html>"))
        self.lineEdit_password.setToolTip(_translate("SignUpPage", "Password must contain atleast 8 characters.Spaces not allowed."))
        self.pushButton_create.setText(_translate("SignUpPage", "C R E A T E"))
        self.pushButton_back.setText(_translate("SignUpPage", "< B A C K"))
        self.label_caa.setText(_translate("SignUpPage", "<html><head/><body><p><span style=\" font-size:20pt; font-weight:600; color:#ffffff;\">Create An Account</span></p></body></html>"))
        self.label_cpassword.setText(_translate("SignUpPage", "<html><head/><body><p><span style=\" font-size:12pt; color:#ffffff;\">Confirm Password </span></p></body></html>"))
        self.label_dob.setText(_translate("SignUpPage", "<html><head/><body><p><span style=\" font-size:12pt; color:#ffffff;\">Date Of Birth</span></p></body></html>"))
        self.label_phone.setText(_translate("SignUpPage", "<html><head/><body><p><span style=\" font-size:12pt; color:#ffffff;\">Phone Number</span></p></body></html>"))
        self.label_H_archive.setText(_translate("SignUpPage", "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.0//EN\" \"http://www.w3.org/TR/REC-html40/strict.dtd\">\n"
"<html><head><meta name=\"qrichtext\" content=\"1\" /><style type=\"text/css\">\n"
"p, li { white-space: pre-wrap; }\n"
"</style></head><body style=\" font-family:\'Berlin Sans FB\'; font-size:8pt; font-weight:400; font-style:normal;\">\n"
"<p style=\" margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><span style=\" font-family:\'Calibri\'; font-size:18pt; color:#33df33;\"> </span><span style=\" font-size:18pt; color:#ffffff;\">Λ</span><span style=\" font-family:\'Calibri (body)\'; font-size:18pt; color:#ffffff;\"> R </span><span style=\" font-family:\'Calibri (Body)\'; font-size:18pt; color:#ffffff;\">C H I V Ξ </span><span style=\" font-family:\'Calibri\'; font-size:18pt; color:#ffffff; vertical-align:super;\">™</span><span style=\" font-family:\'Calibri\'; font-size:18pt;\">             </span></p></body></html>"))
        self.bg_img.setWhatsThis(_translate("SignUpPage", "Password must contain atleast 8 characters"))
        self.label_H_password.setText(_translate("SignUpPage", "<html><head/><body><p><span style=\" font-family:\'Calibri\'; font-size:20pt; font-weight:600; color:#24f100;\">Password</span></p></body></html>"))
